packages/react-aria-components/src/Table.tsx

```diff
@@ -230,13 +230,62 @@ function ResizableTableContainer(props: ResizableTableContainerProps, ref: Forwa
     onResizeEnd: props.onResizeEnd
   }), [width, props.onResizeStart, props.onResize, props.onResizeEnd]);
 
+  // Ensure that the row containing focus is not obscured by a thead with position: sticky.
+  const [isSticky, setIsSticky] = useState<boolean | undefined>(undefined);
+  const [scrollPaddingTop, setScrollPaddingTop] = useState<string | undefined>(undefined);
+  const requestRef = useRef<number | undefined>(undefined);
+  useEffect(() => {
+    if (isSticky === undefined) {
+      // Wait a frame to getComputedStyle from sticky thead.
+      requestRef.current = requestAnimationFrame(() => {
+        // When the thead is sticky, we need to set scroll-padding-top on the container to ensure that a focused row is not obscured but the sticky thead.
+        const thead = objectRef.current?.querySelector('thead, [role="rowgroup"]') as HTMLElement;
+        if (thead && scrollPaddingTop === undefined) {
+          const isTheadSticky = getComputedStyle(thead).position === 'sticky';
+          setIsSticky(isTheadSticky);
+          if (isTheadSticky) {
+            const theadOffsetHeight = thead.offsetHeight;
+            if (CSS.supports('scroll-padding-top', `${theadOffsetHeight}px`)) {
+              setScrollPaddingTop(`${theadOffsetHeight}px`);
+            }
+          }
+        }
+      });
+    }
+    return () => {
+      if (requestRef.current !== undefined) {
+        cancelAnimationFrame(requestRef.current);
+      }
+    };
+  }, [objectRef, isSticky, scrollPaddingTop]);
```

Comment:
> Do we need both solutions? I thought the CSS one would be enough. Or were you just demonstrating?
I think the javascript solution, or css added with javascript as done here, would be more robust. With users bringing their own CSS, I'm not sure we can count on them adding `scroll-padding` to the container when they add `sticky` to the `thead`, otherwise we would never have discovered this bug. We may not need the `onFocusCapture` fallback with adequate browser support for `scroll-padding`.

Reply:
Note: I removed `onFocusCapture` fallback.

Reply:
Didn't realize this was in the component itself. I was thinking of just solving the immediate issue in the example. I'm not sure what the side effects of us adding scroll-padding automatically would be. If we did do that we'd probably want to do it somewhere deeper as this would affect more than just table (e.g. if you have sticky section headers in a menu or another collection).

Reply:
I can remove the change to the component itself and just make the change in the documentation example, but my concern about the potential for error with sticky still stands.